apiserver/networks/state.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package networks
+
+import (
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/state"
+)
+
+// StatePool provides the subset of a state pool required by the
+// remote firewaller facade.
+type StatePool interface {
+	// Get returns a State for a given model from the pool.
+	Get(modelUUID string) (State, func(), error)
+}
+
+// State provides the subset of global state required by the
+// remote firewaller facade.
+type State interface {
+	// ModelUUID returns the model UUID for the model
+	// controlled by this state instance.
+	ModelUUID() string
+
+	// WatchSubnets returns a StringsWatcher that notifies of changes to
+	// the lifecycles of the subnets in the model.
+	WatchSubnets() state.StringsWatcher
+}
+
+type statePoolShim struct {
+	*state.StatePool
+}
+
+func (pool statePoolShim) Get(modelUUID string) (State, func(), error) {
+	st, closer, err := pool.StatePool.Get(modelUUID)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+	return stateShim{st}, closer, nil
+}
+
+type stateShim struct {
+	*state.State
+}
+
+func (st stateShim) WatchSubnets() state.StringsWatcher {
```

Comment:
this is unnecessary, since state.State already has the WatchSubnets method with the required signature